var/spack/repos/builtin/packages/xgboost/package.py

```diff
@@ -0,0 +1,97 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+import shutil
+import subprocess
+import glob
+
+
+class Xgboost(Package):
+    """
+    XGBoost is an optimized distributed gradient boosting library designed to
+    be highly efficient, flexible and portable. It implements machine learning
+    algorithms under the Gradient Boosting framework. XGBoost provides a
+    parallel tree boosting (also known as GBDT, GBM) that solve many data
+    science problems in a fast and accurate way. The same code runs on major
+    distributed environment (Hadoop, SGE, MPI) and can solve problems beyond
+    billions of examples.
+    """
+
+    homepage = "http://xgboost.readthedocs.io/en/latest/"
+    url      = "https://github.com/dmlc/xgboost"
+
+    version('0.7', commit='4aa346c', git="https://github.com/dmlc/xgboost", submodules=True)
+
+    variant('jvm-packages', default=False,
+            description='jvm-packages are compiled')
+
+    variant('gpu', default=False,
+            description='compiled with GPU support')
+
+    depends_on('cmake', type='build')
+    depends_on('maven', type='build', when='+jvm-packages')
+    depends_on('jdk', type='build', when='+jvm-packages')
+    depends_on('cuda', type='build', when='+gpu')
```

Comment:
Don’t you link against it?

Reply:
Good point. Cuda should be also a link dependency.  Not sure about jdk. JDK would be a run/link dependency for whatever software is using *.jar files from +jvm variant, such as grnboost.